### ISSUE TITLE
A few miscellaneous questions

### ISSUE THREAD

**Issue by Dahn-YoungDong** [wpeterman/ResistanceGA#45]
Dear Bill,

I am running resistanceGA on a extent of 2 million cells with 42 populations. There are 8 continuous environmental layers.
The all_comb is a great function that does bootstrap and has a max.combinations function. If I just want single layers that are bootstrapped, should I specify max.combinations as 0? 

A few other questions related to speed and success of optimization. I use lab server in Ubuntu system with 36 CPUS, but it always breaks on me randomly. I suspect it is running out of RAM. 

I am transferring my project to high-performance computing (HPC) cluster, but do not know the benchmark of ResistanceGA? Is more CPU and more RAM linearly increase the computational power to reach the end of optimization? What is the relationship between CPU and RAM here, and should I limit or increase RAM limit? If I had 200 CPUs should I request all that CPU? What is a good way to install julia in HPC and being called succesfully in your wrapper function?

Another question is about GA.prep and jl.prep. What are acceptable ways to limit the transformation parameters for continous variables yet still get acceptable results? Both functions have parallel flags, can they be used in tandem? Does that mean GA.prep will be using separate CPUs as the jl.prep? Should I be cautious of the total CPUs they are divvying up?  Is it okay to change pop.size flag to speed up the optimization process and would it alter too much of the results? What is the default pop.size and can I do pop.size = 10?

Thank you!

**Comment by Dahn-YoungDong** [comment]
Another question, if my all.comb function did not finish so no R object is created, but the results.dir contained partial outputs, is there a way to conduct resist.boot using those partial model-specific outputs, instead of the R object of a succesful all.comb? 
In your example code : mat.list <- c(SS_RESULTS.gdist$cd,
              Multi.Surface_optim.gd$cd,
              Multi.Surface_optim.gd2$cd)
k <- rbind(SS_RESULTS.gdist$k,
           Multi.Surface_optim.gd$k,
           Multi.Surface_optim.gd2$k)

I don't have these R objects saved up due to interrupted runs, but can I extract "cd" from the layerName_jlResistMat.csv file, and K from the All_Results_Table.csv under "K" column?

**Comment by wpeterman** [comment]
I don't have great insight about strategy for installing on HPC and getting things to communicate. Every cluster seems unique, and I haven't endeavored to do this in quite a while. The primary consumption of RAM is the calculation of resistance distance in Julia. It may be more streamlined for you to try using `radish` on your HPC (https://github.com/nspope/radish).

Regarding parallel flags, **only the GA.prep flags will have any effect**. I have not removed those from Julia, but they aren't operational.

There are no right or wrong ways to create continuous landscape variables. Decisions of how to go about doing this may be informed by a variety of things. Just be consistent and have justification for any scaling/resampling done.

Changing the `pop.size` parameter decreases the time to complete each iteration, but you are creating a smaller pool of individuals for selection to act on. You can certainly experiment with this to see how/if it changes results.

If wanting to bootstrap single layer results only, run `SS_optim`, then manually put together the necessary pieces needed to run the `Resist.boot` function, as referenced here: [https://github.com/wpeterman/ResistanceGA/issues/42#issuecomment-1924198753](url)

**Comment by Dahn-YoungDong** [comment]
Hello Bill,

Thank you for your responses. I'll be presenting results from resistanceGA at a conference, though preliminary with limited optimizations. 

In the Run_CS.jl function, how do I specify flags to speed up the circuitscape current map generation process? The only one I can see is the neighbor.connect flag from 8 to 4. Circuitscape jl website says I can also choose all-to-one mode instead all pairwise mode. But I don't see a flag in the resistanceGA function to specify these things 
```
jl_inputs <- jl.prep(n.Pops = length(spatialpointdataframe),
                     CS_Point.File = spatialpointdataframe,
                     response = gdist,
                     JULIA_HOME = JULIA_HOME,
                     parallel = TRUE,
                     cores = 8,
                     Neighbor.Connect = 4,
                     run_test = FALSE,
                     write.files = "circuitscape_connectivity")


Run_CS.jl(jl.inputs = jl_inputs,
          r =  "XX.asc",
          CurrentMap = TRUE,
          full.mat = FALSE,
          EXPORT.dir = "circuitscape_connectivity",
          output = "raster",
          CS_Point.File = spatialpointdataframe,
          pairs_to_include = NULL,
          pop2ind = NULL,
          parallel = FALSE, 
          cores = NULL,
          cholmod = TRUE,
          precision = FALSE, 
          JULIA_HOME = NULL,
          Julia_link = NULL,
          rm.files = TRUE,
          scratch = NULL,
          is_resistance = TRUE)
```

**Comment by wpeterman** [comment]
Set `precision = TRUE`

Review the Details of the `julia.prep` function. If precision is TRUE, then the EXPERIMENTAL single precision method will be used. Single precision usually uses less memory, but is likely to reduce accuracy. NOTE: Preliminary testing of single precision mode in a Windows pc resulted in extremely slow runs.

**Comment by aaabid** [comment]
Hi, 
I have a quick question about the package. My goal is to assess how IBE, IBD, and IBR correlate with genetic data (Fst). I already have the IBE and IBD matrices but still need the resistance matrix (IBR). I previously used Circuitscape 4.0 (Not through resistanceGA but GUI) to generate pairwise resistance matrices for Mantel tests or MMMR, but it keeps failing with errors on different PCs.
While resistanceGA identifies the best landscape feature explaining genetic variation, my main focus is still comparing the relative influence of all three matrices. Can I use resistanceGA to get the resistance matrix for this purpose? I see pairwise matrix outputs csv files but not sure if this can be used, and couldn’t find references for using them in Mantel/MMMR. What’s your advice? Thanks

**Comment by wpeterman** [comment]
> Hi, I have a quick question about the package. My goal is to assess how IBE, IBD, and IBR correlate with genetic data (Fst). I already have the IBE and IBD matrices but still need the resistance matrix (IBR). I previously used Circuitscape 4.0 (Not through resistanceGA but GUI) to generate pairwise resistance matrices for Mantel tests or MMMR, but it keeps failing with errors on different PCs. While resistanceGA identifies the best landscape feature explaining genetic variation, my main focus is still comparing the relative influence of all three matrices. Can I use resistanceGA to get the resistance matrix for this purpose? I see pairwise matrix outputs csv files but not sure if this can be used, and couldn’t find references for using them in Mantel/MMMR. What’s your advice? Thanks

ResistanceGA can be used to optimize a resistance surface, after which a pairwise distance matrix is created. This distance matrix can be used with the `mlpe_rga` function to fit an MLPE model. You can assess level of support for your IBE, IBD, and IBR hypotheses by comparing AIC and/or marginal R^2 of teh fitted MLPE models.

**Comment by aaabid** [comment]
Thanks for your response. 
Just to clarify, are you saying that the matrices I obtained (e.g., elev.lc_commuteDistance_distMat.csv) for elevation and landscape from the optimization process can be used in the same way as matrices typically generated by Circuitscape? 
If so, I would called this distance as IBR and simply fit the model to check its influence on FST Matrices, using either MLPE, or Mantel/MMR see

MLPE.lmm(resistance, 
                pairwise.genetic, 
                REML = FALSE, 
                ID = NULL, 
                ZZ = NULL,
                scale = TRUE)

Thanks!

**Comment by wpeterman** [comment]
Yes, commute distance from gdistance is the same resistance distance from circuitscape.

**Comment by aaabid** [comment]
Quite clear now, and thank you :)
